### ISSUE TITLE
Cannot set the okta_policy_password password_history_count to 0

### ISSUE THREAD

**Issue by michaeldop** [okta/terraform-provider-okta#1147]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```shell
$ terraform -v                                                        
Terraform v1.2.1
on darwin_amd64
+ provider registry.terraform.io/okta/okta v3.28.0
```

### Affected Resource(s)

- okta_policy_password

### Terraform Configuration Files

```hcl
data "okta_group" "all" {
  name = "Everyone"
}

resource "okta_policy_password" "test" {
  name                   = "testAcc_replace_with_uuid"
  status                 = "ACTIVE"
  description            = "Terraform Acceptance Test Password Policy"
  password_history_count = 0
  groups_included        = [data.okta_group.all.id]
}
```

### Debug Output

```
Terraform will perform the following actions:

  # okta_policy_password.test will be updated in-place
  ~ resource "okta_policy_password" "test" {
        id                                     = "00p18j8u01q3eQ7dQ0h8"
        name                                   = "testAcc_replace_with_uuid"
      ~ password_history_count                 = 4 -> 0
        # (26 unchanged attributes hidden)
    }
```

### Expected Behavior

After running apply I would expect to see the `password_history_count` set to 0 instead of 4 inside of okta. When I run plan again it see that the value is still 4.

### Actual Behavior

`password_history_count` is set to 0

### Steps to Reproduce

1. `terraform apply`
1. `terraform plan`

### References
I think its related to this PR https://github.com/okta/terraform-provider-okta/pull/1108

**Comment by monde** [comment]
Thanks @mdop-wh that seems like an odd bug. I checked the api documentation to be sure there wasn't drift on the meaning of the history count being 0 to reset, that is still the same https://developer.okta.com/docs/reference/api/policy/#age-object . And, our integration test `TestAccOktaPolicyPassword_crud` which changes `password_history_count` from 4 to 5 also still passes. The bug, if it exists, doesn't jump out at me after a quick look at the code.
```
$ TF_ACC=1 go test -tags unit -mod=readonly -test.v -run ^TestAccOktaPolicyPassword_crud$ ./okta 2>&1
=== RUN   TestAccOktaPolicyPassword_crud
--- PASS: TestAccOktaPolicyPassword_crud (11.51s)
PASS
ok      github.com/okta/terraform-provider-okta/okta    12.047s
```

**Comment by pro4tlzz** [comment]
I have this problem as well

**Comment by OnwardKnight** [comment]
> Thanks @mdop-wh that seems like an odd bug. I checked the api documentation to be sure there wasn't drift on the meaning of the history count being 0 to reset, that is still the same https://developer.okta.com/docs/reference/api/policy/#age-object . And, our integration test `TestAccOktaPolicyPassword_crud` which changes `password_history_count` from 4 to 5 also still passes. The bug, if it exists, doesn't jump out at me after a quick look at the code.

I just ran into this bug this week @monde, the problem appears to be exclusive to setting `password_history_count` to 0. All other valid integers work.

**Comment by CarsonTolleshaug** [comment]
Any update here? We're also experiencing this issue, seems like it should be an easy fix right?

I would assume the issue is that 0 is also the default for an integer, and so the property is assumed to not be set and thus the default behavior (which is to set password history to 4) is performed. You just need to add someway of distinguishing between setting the property to 0 and not setting it at all. If there's a way to make the property a nullable integer that would probably do the trick.

**Comment by cassdeckard** [comment]
> And, our integration test `TestAccOktaPolicyPassword_crud` which changes `password_history_count` from 4 to 5 also still passes. 

If you modify the `basic_updated.tf` in that test directory to set `password_history_count` to `0` and modify the test expectation accordingly, you get this error:

```
$ TF_ACC=1 go test -tags unit -mod=readonly -test.v -run ^TestAccOktaPolicyPassword_crud$ ./okta/
=== RUN   TestAccOktaPolicyPassword_crud
    resource_okta_policy_password_test.go:21: Step 2/2 error: Check failed: Check 16/24 error: okta_policy_password.test: Attribute 'password_history_count' expected "0", got "4"
--- FAIL: TestAccOktaPolicyPassword_crud (39.15s)
FAIL
FAIL	github.com/okta/terraform-provider-okta/okta	39.413s
FAIL
```

**Comment by monde** [comment]
Okta internal reference: https://oktainc.atlassian.net/browse/OKTA-547413
Schedule for work in our next sprint, fyi.

**Comment by monde** [comment]
Note to self: this is on deck. It is a JSON rendering issue on okta-sdk-golang's use of omitempty and history count being set to zero.

**Comment by bcoquard** [comment]
Hello,

Is the problem fixed somewhere ?
I am having the exact same issue. I am using Okta Terraform : version = "3.41.0" (currently the latest version).

In my config file.

I am setting with module okta_policy_password : 
`password_history_count = 0`

Once terraform plan is ran I get the output :
```
~ resource "okta_policy_password" "this" {
        id                                     = "..."
        name                                   = "..."
      ~ password_history_count                 = 4 -> 0
        # (28 unchanged attributes hidden)
    }
```

But once the plan is applied nothing happens I get stuck with this password_history_count default value of 4.

Any idea what needs to be fixed ?

Thank you.

**Comment by monde** [comment]
@bcoquard Not being able to set the count to zero is still a bug, we haven't fixed it yet.

**Comment by bahadirh** [comment]
@monde @pro4tlzz
I've noticed that this bug is related to [one of your internal packages](https://github.com/okta/okta-sdk-golang) and it's already solved [in this PR](https://github.com/okta/okta-sdk-golang/pull/350/files#diff-3b88a1cae5eb3dfb7c55a3f91977e225443f21d5db87e1f7fa33207267ca1a78R23). You just have to dump `okta-sdk-golang`'s version. `v2.16.0` is, I believe, the first version that includes the solution.

**Comment by monde** [comment]
@bahadirh just bumping the okta-sdk-golang version to get that change I created is too risky to be 100% certain we aren't introducing subtle bugs across the provider with the change.

okta-sdk-golang is getting a major version release where we have a new openapi spec that all Okta SDKs are generated from. For golang, all fields will be field pointers. Therefore, we have decided to take on all that risk in a coming major version release for the Okta provider. This release is our current Q1 goal for this year.
